From c7159f1defaf4a0dc5cdd191c4dadaa6c83dcd0b Mon Sep 17 00:00:00 2001
From: Lukasz <lukaszunity@users.noreply.github.com>
Date: Fri, 10 Jul 2015 10:27:55 +0200
Subject: [PATCH] Put quotes around path to mono. Fixes bug where Process.Start
 will fail for a managed process if the path to mono contains spaces.

---
 mono/io-layer/processes.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mono/io-layer/processes.c b/mono/io-layer/processes.c
index a57d8aa..e806bd8 100644
--- a/mono/io-layer/processes.c
+++ b/mono/io-layer/processes.c
@@ -815,11 +815,11 @@ gboolean CreateProcess (const gunichar2 *appname, const gunichar2 *cmdline,
 
 		if (newapp != NULL) {
 			if (appname != NULL) {
-				newcmd = utf16_concat (newapp, utf16_space,
+				newcmd = utf16_concat (utf16_quote, newapp, utf16_quote, utf16_space,
 						       appname, utf16_space,
 						       cmdline, NULL);
 			} else {
-				newcmd = utf16_concat (newapp, utf16_space,
+				newcmd = utf16_concat (utf16_quote, newapp, utf16_quote, utf16_space,
 						       cmdline, NULL);
 			}
 			
-- 
2.2.1

